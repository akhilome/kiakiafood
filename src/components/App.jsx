import '../index.css';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import propTypes from 'prop-types';
import { checkAuthStatus } from '../actions';
import HomePage from './HomePage';
import LoginPage from './LoginPage';
import Signup from './SignupPage';
import NotFoundPage from './NotFoundPage';

export class App extends Component {
  componentDidMount() {
    const { checkAuthStatus: checkUserAuthStatus } = this.props;
    checkUserAuthStatus();
  }

  render() {
    return (
      <Router>
        <Switch>
          <Route path="/" exact component={HomePage} />
          <Route path="/login" component={LoginPage} />
          <Route path="/signup" component={Signup} />
          <Route component={NotFoundPage} />
        </Switch>
      </Router>
    );
  }
}

App.propTypes = {
  checkAuthStatus: propTypes.func.isRequired,
};

export default connect(
  null,
  { checkAuthStatus },
)(App);
