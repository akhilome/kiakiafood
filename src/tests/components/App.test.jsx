import React from 'react';
import { shallow } from 'enzyme';
import App from '../../components/App';

describe('App component', () => {
  it('should render app correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
