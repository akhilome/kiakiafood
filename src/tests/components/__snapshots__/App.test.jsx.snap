// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render app correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <App
    checkAuthStatus={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    getCart={[MockFunction]}
    isLoggedIn={false}
    loading={false}
  >
    <BrowserRouter>
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Connect(Nav)>
          <Nav
            cartItemsCount={0}
            dispatch={[Function]}
            isLoggedIn={null}
            role=""
          >
            <header>
              <div
                className="site-title"
              >
                <h2>
                  <Link
                    replace={false}
                    to="/"
                  >
                    <a
                      href="/"
                      onClick={[Function]}
                    >
                      Kiakia Food
                    </a>
                  </Link>
                </h2>
              </div>
              <nav>
                <ul>
                  <li>
                    <Link
                      replace={false}
                      to="/login"
                    >
                      <a
                        href="/login"
                        onClick={[Function]}
                      >
                        Log In
                      </a>
                    </Link>
                  </li>
                  <li>
                    <Link
                      replace={false}
                      to="/signup"
                    >
                      <a
                        href="/signup"
                        onClick={[Function]}
                      >
                        Sign Up
                      </a>
                    </Link>
                  </li>
                </ul>
              </nav>
            </header>
          </Nav>
        </Connect(Nav)>
        <Switch>
          <Route
            component={[Function]}
            computedMatch={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            exact={true}
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            path="/"
          >
            <HomePage
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
            >
              <section
                className="hero"
              >
                <h2>
                  Quell Your Hunger in Minutes
                </h2>
                <Link
                  replace={false}
                  to="/menu"
                >
                  <a
                    href="/menu"
                    onClick={[Function]}
                  >
                    <button
                      className="btn-primary"
                      type="button"
                    >
                      Get Food ‚ü∂
                    </button>
                  </a>
                </Link>
              </section>
            </HomePage>
          </Route>
        </Switch>
      </Router>
    </BrowserRouter>
  </App>
</Provider>
`;

exports[`App component should render app correctly for authenticated customers 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <App
    checkAuthStatus={
      [MockFunction] {
        "calls": Array [
          Array [],
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    getCart={[MockFunction]}
    isLoggedIn={true}
    loading={false}
  >
    <BrowserRouter>
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Connect(Nav)>
          <Nav
            cartItemsCount={0}
            dispatch={[Function]}
            isLoggedIn={null}
            role=""
          >
            <header>
              <div
                className="site-title"
              >
                <h2>
                  <Link
                    replace={false}
                    to="/"
                  >
                    <a
                      href="/"
                      onClick={[Function]}
                    >
                      Kiakia Food
                    </a>
                  </Link>
                </h2>
              </div>
              <nav>
                <ul>
                  <li>
                    <Link
                      replace={false}
                      to="/login"
                    >
                      <a
                        href="/login"
                        onClick={[Function]}
                      >
                        Log In
                      </a>
                    </Link>
                  </li>
                  <li>
                    <Link
                      replace={false}
                      to="/signup"
                    >
                      <a
                        href="/signup"
                        onClick={[Function]}
                      >
                        Sign Up
                      </a>
                    </Link>
                  </li>
                </ul>
              </nav>
            </header>
          </Nav>
        </Connect(Nav)>
        <Switch>
          <Route
            component={[Function]}
            computedMatch={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            exact={true}
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            path="/"
          >
            <HomePage
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
            >
              <section
                className="hero"
              >
                <h2>
                  Quell Your Hunger in Minutes
                </h2>
                <Link
                  replace={false}
                  to="/menu"
                >
                  <a
                    href="/menu"
                    onClick={[Function]}
                  >
                    <button
                      className="btn-primary"
                      type="button"
                    >
                      Get Food ‚ü∂
                    </button>
                  </a>
                </Link>
              </section>
            </HomePage>
          </Route>
        </Switch>
      </Router>
    </BrowserRouter>
  </App>
</Provider>
`;
